jcstress-samples/src/main/java/org/openjdk/jcstress/samples/concurreny/mutex/Mutex_03_AtomicBoolean.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jcstress.samples.concurreny.mutex;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.II_Result;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE_INTERESTING;
+
+/*
+    How to run this test:
+        $ java -jar jcstress-samples/target/jcstress.jar -t Mutex_03_AtomicBoolean
+*/
+
+@JCStressTest
+@Outcome(id = {"1, 2", "2, 1"}, expect = ACCEPTABLE, desc = "Sequential execution.")
+@Outcome(id = "1, 1", expect = ACCEPTABLE_INTERESTING, desc = "Both actors came up with the same value: lock failure.")
+@State
+public class Mutex_03_AtomicBoolean {
+    private final AtomicBoolean taken = new AtomicBoolean(false);
+    private int v;
+
+    @Actor
+    public void actor1(II_Result r) {
+        while(taken.get() || !taken.compareAndSet(false, true)); // spin
```

Comment:
```suggestion
        while(!taken.compareAndSet(false, true)) {
            // wait for entering into the critical section
        }
```

I think `taken.get()` is not needed here, and adding explicit braces makes it easier to understand that the braces for "critical section" are different ones.

Reply:
Well, @shipilev proposed it. I cannot find his comment.

Reply:
This is TTAS, it is OK in this form.